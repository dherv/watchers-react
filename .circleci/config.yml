version: 2
jobs:
  build:
    working_directory: ~/tmp
    branches:
      only:
        - develop
        - feature/*
        - master
    docker:
      - image: circleci/node
    steps:
      - checkout
      - run: yarn install
      - run: yarn test
