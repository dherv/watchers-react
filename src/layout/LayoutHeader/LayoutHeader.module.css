.header {
  display: flex;
  align-items: center;
  margin-bottom: 0rem;
  background-color: #181717;
  box-shadow: var(--shadow);
}
