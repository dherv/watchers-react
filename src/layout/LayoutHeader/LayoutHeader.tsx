import React, { PureComponent } from "react";
import styled from "styled-components";
import LayoutNav from "../LayoutNav/LayoutNav";
import Logo from "../../components/Logo";

// ---------------
interface IProps {}

export default class LayoutHeader extends PureComponent {
  render() {
    return (
      <Header>
        <Logo />
        <LayoutNav />
      </Header>
    );
  }
}
// ---------------
const Header = styled.header`
  display: flex;
  align-items: center;
  padding: 0 4rem;
  margin-bottom: 0rem;
  background-color: #181717;
  box-shadow: var(--shadow);
`;
