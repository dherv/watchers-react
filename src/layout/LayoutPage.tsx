import React, { PureComponent } from "react";
import styled from "styled-components";
import Header from "./LayoutHeader/LayoutHeader";

export default class LayoutPage extends PureComponent {
  render() {
    return (
      <Page>
        <Header />
        <Main>{this.props.children}</Main>
      </Page>
    );
  }
}
const Page = styled.div``;
const Main = styled.main`
  padding: 4rem;
`;
