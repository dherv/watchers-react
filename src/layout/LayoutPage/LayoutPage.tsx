import React, { Component } from "react";
import LayoutHeader from "../LayoutHeader/LayoutHeader";
import styles from "./LayoutPage.module.css";

export default class LayoutPage extends Component {
  render() {
    return (
      <div>
        <LayoutHeader />
        <main className={styles.main}>{this.props.children}</main>
      </div>
    );
  }
}
