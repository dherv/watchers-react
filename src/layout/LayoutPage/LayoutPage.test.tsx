import { renderer, shallow, React } from "../../test.config";
import LayoutPage from "./LayoutPage";
import { MemoryRouter } from "react-router";
import LayoutHeader from "../LayoutHeader/LayoutHeader";
describe("LayoutPage", () => {
  const props = {
    children: "children"
  };
  const wrapper = shallow(<LayoutPage {...props} />);
  describe("component", () => {
    it("should display one <div>", () => {
      expect.assertions(1);
      expect(wrapper.find("div")).toHaveLength(1);
    });
    it("should display one <LayoutHeader>", () => {
      expect.assertions(1);
      expect(wrapper.find(LayoutHeader)).toHaveLength(1);
    });
    it("should display one <main>", () => {
      expect.assertions(1);
      expect(wrapper.find("main")).toHaveLength(1);
    });
    it("should display one <main> with prop its children equal to props.children", () => {
      expect.assertions(1);
      expect(wrapper.find("main").prop("children")).toEqual(props.children);
    });
  });
  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const tree = renderer
        .create(
          <MemoryRouter>
            <LayoutPage {...props} />
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
