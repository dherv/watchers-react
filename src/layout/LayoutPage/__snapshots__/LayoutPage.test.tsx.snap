// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutPage snapshot should match 1`] = `
<div>
  <header
    className="header"
  >
    <h2
      className="logo"
    >
      Watchers
    </h2>
    <nav
      className="nav"
    >
      <ul
        className="list"
      >
        <li>
          <a
            aria-current={null}
            className="link"
            href="/movies"
            onClick={[Function]}
          >
            movies
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="link"
            href="/series"
            onClick={[Function]}
          >
            series
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="link"
            href="/likes"
            onClick={[Function]}
          >
            likes
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="link"
            href="/watchlist"
            onClick={[Function]}
          >
            watchlist
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="link"
            href="/account"
            onClick={[Function]}
          >
            account
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    children
  </main>
</div>
`;
